src/HappyConnOpener.cc

```diff
@@ -397,15 +397,11 @@ HappyConnOpener::swanSong()
     // TODO: Find an automated, faster way to kill no-longer-needed jobs.
 
     if (prime) {
-        if (prime.connector)
-            prime.connector->cancel("HappyConnOpener object destructed");
-        prime.clear();
+        cancelAttempt(prime, "HappyConnOpener object destructed");
```

Comment:
Rule of thumb: Avoid identical description of different contexts. Unique descriptions simplify triage.

```suggestion
        cancelAttempt(prime, "job finished during a prime attempt");
```